setup/migrate.py

```diff
@@ -191,6 +191,62 @@ def migration_14(env):
 	db = os.path.join(env["STORAGE_ROOT"], 'mail/users.sqlite')
 	shell("check_call", ["sqlite3", db, "CREATE TABLE auto_aliases (id INTEGER PRIMARY KEY AUTOINCREMENT, source TEXT NOT NULL UNIQUE, destination TEXT NOT NULL, permitted_senders TEXT);"])
 
+def migration_15(env):
+	backup_configuration_file = os.path.join(env["STORAGE_ROOT"], 'backup', 'custom.yaml')
+	if os.path.exists(backup_configuration_file):
+		import yaml
+
+		with open(backup_configuration_file, 'r') as f:
+			custom_config = yaml.load(f, Loader=yaml.SafeLoader)
+		if not isinstance(custom_config, dict): raise ValueError()
```

Comment:
This is much better and makes backup.py a lot more readable imo. 

One thing is that the rest of the migration functions don't throw errors,  they just print errors. This seems like a good decision:  there's a later `except.. sys.exit` so if something strange happens here it could leave the system in a broken state. Considering this is non-critical functionality I don't think we have to do that; it's probably acceptable to recover gracefully by doing something like renaming custom.yaml to custom.backup.yaml, printing an error indicating that backup config was corrupt and has been renamed and returning early. 

I'd maybe also check whether `custom_config["type"]` is already set and likewise print a warning and return early; it shouldn't be the case that the migration runs twice, but, if it does it's probably best not to silently corrupt the configuration (by e.g. overwriting target_url (existing) with target (not existing).)

Reply:
Solved with c2636f68f898fd77368350ae2039b08e47de33b3.